### DISCUSSION TITLE
Serializer [normalizer] - Mapping via attributes

### DISCUSSION THREAD

**Discussion by florianhuber81** [symfony/symfony#45425] [Q&A]
Any plans to go from this:

`    /**
     * @param ArrayCollection<\App\Model\Absence> $absences
     */
    public function setAbsences(ArrayCollection $absences) : void
    {
        $this->absences = $absences;
    }`

to something like this:
`    #[mapping([ArrayCollection::class => Absence::class])]
    public function setAbsences(ArrayCollection $absences) : void
    {
        $this->absences = $absences;
    }`

Or is that already possible somehow?

**Comment by mbabker** [comment]
Personally, I don't see the point.  Doc blocks are for extra metadata that can't be captured in code, and PHP doesn't have language-level generics, so doc block annotations are perfect for this use case.  Plus, a lot of tooling in the ecosystem (static analyzers, doc block parsers, and IDEs) handle parsing doc blocks for this extra data to make a more cohesive experience.  Moving the mapping in this example to a serializer specific PHP attribute means you're going to lose a lot of the benefit of that tooling unless you also keep the doc block in place, at which point the attribute really doesn't offer much benefit.
